src/all/pornpics/src/eu/kanade/tachiyomi/extension/all/pornpics/PornPics.kt

```diff
@@ -0,0 +1,200 @@
+                }
+            }
+        } else {
+            val doc = response.asJsoup()
+            doc.select(simpleMangaSelector()).map {
+                val imgEl = it.selectFirst("img")!!
+                SManga.create().apply {
+                    setUrlWithoutDomain(it.absUrl("href"))
+                    title = imgEl.attr("alt")
+                    thumbnail_url = imgEl.absUrl("data-src")
+                }
+            }
+        }
+        // response maybe [], Add +1 to requested image count per page,
+        // Compare actual received count with pageSize to determine next page.
+        val hasNextPage = mangas.size > QUERY_PAGE_SIZE
+        val readerMangas = if (hasNextPage) mangas.dropLast(1) else mangas
+        return MangasPage(readerMangas, hasNextPage)
+    }
+
+    override fun simpleNextPageSelector(): String? = null
+
+    override fun popularMangaRequest(page: Int) = buildMangasPageRequest(page, 1)
+
+    override fun latestUpdatesRequest(page: Int) = buildMangasPageRequest(page, 2)
+
+    private fun buildMangasPageRequest(page: Int, period: Int): Request {
+        val categoryOption = PornPicsPreferences.getCategoryOption(getPreferences())
+        val isDefaultCategory = PornPicsPreferences.DEFAULT_CATEGORY_OPTION == categoryOption
+
+        val builder = when {
+            isDefaultCategory -> "$baseUrl/popular/api/galleries/list"
+            period == 1 -> "$baseUrl/$categoryOption"
+            period == 2 -> "$baseUrl/$categoryOption/recent"
+            else -> baseUrl
+        }.toHttpUrl().newBuilder()
+            .addQueryPageParameter(page)
+            .addQueryParameter("lang", intl.chosenLanguage)
+
+        if (isDefaultCategory) {
+            builder.addQueryParameter("category_id", 2585 + period)
+                .addQueryParameter("period", period)
+        }
+
+        return GET(builder.build(), headers)
+    }
+
+    override fun mangaDetailsParse(document: Document): SManga {
+        val thumbEl = document.selectFirst(simpleMangaSelector())!!
+        val imgEl = thumbEl.selectFirst("img")!!
+        val infoEl = document.selectFirst("div.gallery-info.to-gall-info")
+
+        return SManga.create().apply {
+            title = imgEl.attr("alt")
+            update_strategy = UpdateStrategy.ONLY_FETCH_ONCE
+            status = SManga.COMPLETED
+            author = infoEl?.select("div.gallery-info__item:nth-child(2) a")?.joinToString { it.text() }
+            genre = infoEl?.select("div.gallery-info__item:not(:nth-child(2)) a")?.joinToString { it.text() }
+            description = infoEl?.selectFirst("div.gallery-info__item:nth-child(4)")?.text()
+        }
+    }
+
+    override fun chapterListSelector() = "li.mobile a.alt-lang-item[data-lang=en]"
+    override fun chapterFromElement(element: Element) = SChapter.create().apply {
+        chapter_number = 0F
+        setUrlWithoutDomain(element.absUrl("href"))
+        name = intl["chapter.name.default"]
+    }
```

Comment:
not really using the response here, override `fetchChapterList` and return chapter from there without making network call